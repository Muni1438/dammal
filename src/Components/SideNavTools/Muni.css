.h1Tag{
    color: rgb(23, 141, 141);
    text-decoration: underline;
    text-decoration-color: aquamarine;
}
.inputTag{

    height: 30px;
    width: 200px;
    /* outline-color: orange; */
    border-top: none;
    border-bottom-color: paleturquoise;
    border-left: none;
    border-right: none;
    outline: none;
    font-size: 25px;
    
}
.h2Tag{
    font-size: 30px;
    width: fit-content;
    height: fit-content;
    background-color: rgb(136, 169, 177);
    border: slategrey 2px solid;
    margin:25px auto 25px auto;
    border-radius: 3px;
    color: snow;
    
    padding: 2px 5px 2px 5px;
}