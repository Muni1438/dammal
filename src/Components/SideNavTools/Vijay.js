import React, { useState, useRef } from "react";
import ClassTest from "../ClassTest";
import data from "../dataStorage.json";
function Vijay() {
  const [fetching, setfetching] = useState(0);
  const [Focus, setFocus] = useState("");
  let AllAbout = useRef();
  const ForInput = useRef();
  if (fetching < data.length) {
    AllAbout = new Array(1).fill(data[fetching]);
  }
  const ClickToFocus = () => {
    ForInput.current.focus();
  };
  return ( 
    <div>
      <button onClick={() => setfetching(fetching + 1)}>Fetch A Movie</button>
      {console.log(AllAbout)}
      {AllAbout.length ? (
        AllAbout.map((mine) => (
          <div key={mine.id}>
            <h3>Hero: {mine.hero}</h3>
            <h1>Movie: {mine.movie}</h1>
          </div>
        ))
      ) : ( 
        <h1>No Movies Left To Fetch</h1>
      )}
      <input
        type="text"
        value={Focus}
        ref={ForInput}
        onChange={(e) => setFocus(e.target.value)}
      />
      <button onClick={ClickToFocus}>Focus</button>
      <ClassTest/>
    </div>
  );
}

export default Vijay;
