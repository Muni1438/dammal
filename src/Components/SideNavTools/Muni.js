import React, { useState, useReducer } from "react";
import "./Muni.css"
function Muni() {
  const Actions = {
    ADD_TODO: "ADD_TODO",
    DELETE_TODO: "DELETE_TODO",
  };

  const [Text, setText] = useState("");
  const addingAll = (Text) => {
    return {
      id: Math.random(),
      Text: Text,
    };
  };
  const initialState = [];
  const reducer = (state, action) => {
    switch (action.type) {
      case Actions.ADD_TODO:
        return [...state, addingAll(action.payload)];
      case Actions.DELETE_TODO:
        return state.filter((last) => last.id !== action.payload.id);
      default:
        return state;
    }
  };
  const [state, dispatch] = useReducer(reducer, initialState);

  const addingData = () => {
    return {
      type: Actions.ADD_TODO,
      payload: Text,
    };
  };
  const SubmitData = (e) => {
    e.preventDefault();
    dispatch(addingData());
  };
  const deletingData = (Id) => {
    return {
      type: Actions.DELETE_TODO,
      payload: { id: Id },
    };
  };
  return (
    <div>
      <form onSubmit={SubmitData}>
      <h1 className="h1Tag">ToDo App By Muni Kumar</h1>
        <input
          type="text"
          value={Text}
          onChange={(e) => setText(e.target.value)}
          className="inputTag"
          placeholder="/........"
          data-testid="inputTag"
        />
      </form>
      {state ? (
        state.map((data) => (
          <h2 key={data.id} onClick={() => dispatch(deletingData(data.id))} className="h2Tag">
            {data.Text}
          </h2>
        ))
      ) : (
        <h1>Nothing to delete</h1>
      )}
    </div>
  );
}

export default Muni;
