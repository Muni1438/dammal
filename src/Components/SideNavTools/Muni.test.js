import React from 'react'
import {render,cleanup,fireEvent,screen,act } from "@testing-library/react"
import Muni from './Muni';
import Vijay from './Vijay';
describe("All about Todo App",()=>{
afterEach(cleanup);
test('About h1 tag ', () => {
   const{getByTestId,asFragment}= render(<Muni/>)
   expect(asFragment(<Muni/>)).toMatchSnapshot()
    fireEvent.change(getByTestId("inputTag"),{target:{value:"muni kumar"}})
    expect(getByTestId("inputTag").value).toBe("muni kumar")
})
test("about Fetching",async()=>{
    const fetching=jest.fn(()=>Promise.resolve("jakkula"))
    const {asFragment}=render(<Vijay/>)
    expect(asFragment(<Vijay/>)).toMatchSnapshot()


})


})

