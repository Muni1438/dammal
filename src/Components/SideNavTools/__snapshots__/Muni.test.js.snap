// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All about Todo App About h1 tag  1`] = `
<DocumentFragment>
  <div>
    <form>
      <h1
        class="h1Tag"
      >
        ToDo App By Muni Kumar
      </h1>
      <input
        class="inputTag"
        data-testid="inputTag"
        placeholder="/........"
        type="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;

exports[`All about Todo App about Fetching 1`] = `
<DocumentFragment>
  <div>
    <button>
      Fetch A Movie
    </button>
    <div>
      <h3>
        Hero: Ram
      </h3>
      <h1>
        Movie: Rangasthalam
      </h1>
    </div>
    <input
      type="text"
      value=""
    />
    <button>
      Focus
    </button>
  </div>
</DocumentFragment>
`;
